pkg/trtl/honu.go

```diff
@@ -11,15 +14,53 @@ import (
 // A HonuService implements the RPCs for interacting with a Honu database.
 type HonuService struct {
 	pb.UnimplementedTrtlServer
+	parent *Server
+	db     *honu.DB
 }
 
-func NewHonuService() *HonuService {
-	return &HonuService{}
+func NewHonuService(s *Server) (*HonuService, error) {
+	return &HonuService{parent: s, db: s.db}, nil
 }
 
 // Get is a unary request to retrieve a value for a key.
 func (h *HonuService) Get(ctx context.Context, in *pb.GetRequest) (out *pb.GetReply, err error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if in.Options.ReturnMeta {
+		// Retrieve and return the metadata.
+		var object *object.Object
+		if object, err = h.db.Object(in.Key); err != nil {
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
+		out = &pb.GetReply{
+			Value: object.Data,
+			Meta: &pb.Meta{
+				Key:       object.Key,
+				Namespace: object.Namespace,
+				Region:    object.Region,
+				Owner:     object.Owner,
+				Version: &pb.Version{
+					Pid:     object.Version.Pid,
+					Version: object.Version.Version,
+					Region:  object.Version.Region,
+				},
+				Parent: &pb.Version{
+					Pid:     object.Version.Parent.Pid,
+					Version: object.Version.Parent.Version,
+					Region:  object.Version.Parent.Region,
+				},
+			},
+		}
+	} else {
+		// Just return the value for the given key.
+		var value []byte
+		log.Debug().Msg(string(in.Key))
```

Comment:
I do like the idea of adding some log tracing for debugging, but having the log message here is inconsistent since only requests without `ReturnMeta` will log. What if we put the debug message at the top of the request? Also, I think the following format would help us debug things in Google Log explorer a bit easier:

```
log.Debug().Str("key", string(in.Key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Get")
```

Reply:
Oops, I didn't intend to leave that in the PR, I think moving it to the top of the request is a good idea though.